### ISSUE TITLE
[Local] Optimize train.py with custom MLX kernels from newton-mlx

### ISSUE THREAD

**Issue by DiegoLopezInc** [Newton-Group/autoresearch-mlx#3]
## Summary

Integrate custom MLX Metal kernels (from `newton-mlx` repo) into the autoresearch training loop to maximize throughput on Apple Silicon. The 5-minute fixed time budget means faster kernels = more training steps = potentially lower val_bpb.

## Context

- `train.py` uses standard MLX ops (`mx.matmul`, `mx.softmax`, `nn.Linear`, etc.)
- `newton-mlx` will develop custom Metal kernels optimized for M4 Pro/Max architecture
- The fixed 5-min budget makes throughput directly impactful — 2x faster = 2x more steps
- Current best on Mac Mini: 1.353329 val_bpb (46 steps in 312s training)

## Implementation

1. **Fused attention kernel**:
   - Replace standard QKV → matmul → softmax → matmul attention with a single fused Metal kernel
   - Sliding window variant for the "S" layers in window_pattern "SSSL"
   - Integrate TurboQuant 3-bit KV cache compression for memory efficiency

2. **Fused RMSNorm + Linear**:
   - The `norm()` function is called before every attention and FFN block
   - Fuse with the subsequent linear projection to eliminate one memory round-trip

3. **Memory-efficient gradient accumulation**:
   - Custom kernel for in-place gradient accumulation
   - Reduce peak memory from storing full gradient tensors

4. **Integration pattern**:
   ```python
   # In train.py, check for newton-mlx availability
   try:
       from newton_mlx.kernels import fused_attention, fused_norm_linear
       USE_CUSTOM_KERNELS = True
   except ImportError:
       USE_CUSTOM_KERNELS = False
   ```

5. **Benchmark framework**:
   - Before/after comparison: steps completed, val_bpb, peak memory, MFU%
   - Track in `results.tsv` with `[newton-mlx]` prefix in descriptions

## Dependencies
- `newton-mlx` repo custom kernel development
- MLX Metal shader compilation pipeline

## Acceptance Criteria

- [ ] Fused attention kernel integrated and benchmarked
- [ ] Measurable throughput improvement (steps/5min)
- [ ] No val_bpb regression at same step count
- [ ] Graceful fallback when newton-mlx not installed
- [ ] Benchmark results logged to results.tsv

## Labels
`local-pipeline` `mlx` `performance` `newton-mlx`
